app/controllers/admin/drinks_controller.rb

```diff
@@ -0,0 +1,25 @@
+module Admin
+  class DrinksController < ApplicationController
+    before_filter :authenticate_user!
+    before_filter do
```

Comment:
this method is duplicated - how can you use inheritance to maintain a DRY method for authorizing admins?